.github/workflows/publish_prerelease.yml

```diff
@@ -0,0 +1,50 @@
+name: Publish pre-release
+
+on:
+  push:
+    tags:
+      - "pre-release-*"
+
+jobs:
+  code-format:
+    name: Code Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        name: Setup Node.js
+        with:
+          node-version: "18.x"
+      - name: Setup yarn
+        run: npm install -g yarn
+      - run: yarn
+      - run: yarn ci:code-formatting
+  code-style:
+    name: Code Style
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        name: Setup Node.js
+        with:
+          node-version: "18.x"
+      - name: Setup yarn
+        run: npm install -g yarn
+      - run: yarn
+      - run: yarn ci:code-style
+  build:
+    name: build check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+          registry-url: https://registry.npmjs.org
+      - name: Setup yarn
+        run: npm install -g yarn
```

Comment:
we might not need this step `setup yarn` ,as we have in `publish_to_npm.yml` steps. i think `actions/setup-node@v3` already do that for us